bansa/src/main/kotlin/com/brianegan/bansa/RxStore.kt

```diff
@@ -1,31 +1,24 @@
 package com.brianegan.bansa
 
 import rx.Observable
-import rx.Scheduler
 import rx.Subscriber
 import rx.Subscription
-import rx.schedulers.Schedulers
 import rx.subjects.PublishSubject
 import rx.subjects.SerializedSubject
 
 class RxStore<S, A>(
-        private val initialState: S,
-        private val initialReducer: (S, A) -> S,
-        private val scheduler: Scheduler = Schedulers.newThread()
+        override var state: S,
+        val reducer: (S, A) -> S
 ) : Store<S, A> {
-    private val dispatcher: SerializedSubject<A, A>
-    override var state: S = initialState
+
+    private val dispatcher = SerializedSubject<A, A>(PublishSubject.create<A>())
 
     override val stateChanges: Observable<S>
-    var reducer: (S, A) -> S
 
     init {
-        reducer = initialReducer
-        dispatcher = SerializedSubject<A, A>(PublishSubject.create<A>())
         stateChanges = dispatcher // When an action is dispatched
-                .observeOn(scheduler) // Run the scan on a given thread, by default a background thread
```

Comment:
"I think that synchronicity of the stream is one of the core principle of these frameworks"

Yah, that's an interesting point. I don't know if it's a hard requirement that an action have an immediate affect on the state of the app, since any action could be intercepted by middleware and dispatch an asynchronous action instead. But being able to reason about reducers in a synchronous way is a big win, especially at the middleware level. Also, giving people the option to make expensive calculations on the reducer level "since it won't thrash the main thread" might be encouraging the wrong behavior. 

At this point, it's a premature optimization, and if we need to bring it back, it'd be only a few lines of work to do so.

Therefore, let's remove it for now. If a use case comes about where someone really has a good reason to make big computational changes to the state of the app that would require a background thread, we can consider making a change then.

Thanks for letting me ask questions and stew over your changes, even though I end up just accepting them in the end! Really appreciate all the good stuff you've done for the library!